pkg/security/secl/rules/ruleset.go

```diff
@@ -603,13 +604,22 @@ func (rs *RuleSet) GetFieldValues(field eval.Field) []eval.FieldValue {
 
 // IsDiscarder partially evaluates an Event against a field
 func IsDiscarder(ctx *eval.Context, field eval.Field, rules []*Rule) (bool, error) {
+	var isDiscarder bool
+
 	for _, rule := range rules {
+		// ignore rule that can't generate discarders
+		if rule.NoDiscarder {
```

Comment:
Doesn't this mean that we might miss events for `NoDiscarder` rules ? I'm surprised we don't need to check the partial evaluation here.

Reply:
Discussed about it offline. The NoDiscarder flag will be applied for rule having specific approver that will supplant the discarder mechanism kernel side.